src/utilities/createRouteMethods.ts

```diff
@@ -0,0 +1,39 @@
+import { Resolved, Route, RouteMethod, RouteMethodResponse, RouteMethods, Routes, Then, isPublicRoute } from '@/types'
+import { assembleUrl } from '@/utilities/urlAssembly'
+
+export function createRouteMethods<T extends Routes>(routes: Resolved<Route>[]): RouteMethods<T> {
+  const methods = routes.reduce<Record<string, any>>((methods, route) => {
+    let level = methods
+
+    route.matches.forEach(match => {
+      if (!match.name) {
+        return
+      }
+
+      if (match === route.matched && isPublicRoute(route.matched)) {
+        level[route.name] = createCallableNode(route)
+        return
+      }
```

Comment:
Originally this had `Object.assign` with what might exist already at `level[route.name]` but that should never be the case. If we get here we're at a leaf. And the tests all pass so unless we're missing a test I think this is all we need.

Reply:
Just realized I was only looking at the type tests... The unit tests indeed are broken. Looking into that.

Reply:
The `Object.assign` was necessary. Put that back 👍